contracts/utils/math/Math.sol

```diff
@@ -218,6 +218,31 @@ library Math {
         return result;
     }
 
+    /**
+     * @notice Calculate the inverse of a number in Z/nZ.
+     *
+     * If p is a prime, then we Z/nZ is a field commonly noted Fp. In that case all elements are inversible, expect 0.
+     * If p is not a prime, then Z/nZ is not a field, and some elements might not be inversible.
+     *
+     * If the input values is not inversible, 0 is returned.
```

Comment:
I'd suggest using `n` all over:

```suggestion
     * If n is a prime, then Z/nZ is a field commonly noted Fn. In that case all elements are inversible, expect 0.
     * If n is not a prime, then Z/nZ is not a field, and some elements might not be inversible.
     *
     * If the input values is not inversible, 0 is returned.
```

Is there any reason for preferring `p`?

Reply:
Usually, Z/nZ is the group structure defined by the equivalence classes resulting from the modulo operations (and the common addition and multiplication operation). We often use numbers between 0 and n-1 as the canonical elements of each equivalent class.

It turns that that if the characterictic number for this group structure is a prime, then the group is actually a field (all non-zero element are invertible). This is ussually noted as Fp.

So yes, Fp is Z/nZ when n is prime, I've never seen Fn written ... It would be confused with "Function". 
I've seen Z/pZ written, but writting it that way implied the restriction that p is prime.

Reply:
relevant link: https://en.wikipedia.org/wiki/Finite_field

Reply:
Ok, it's clear to me now.

The current version looks good, but feel free to add the "commonly noted as Fp" if you think it's useful.